main/src/main/scala/sbt/Defaults.scala

```diff
@@ -2351,8 +2354,40 @@ object Defaults extends BuildCommon {
         }
       )
     }
+    def onProgress(s: Setup) = {
+      val p = s.progress.asScala
+      s.withProgress {
+        p.collect {
```

Comment:
If for some reason `s.progress` is empty there would be no BSP progress report unfortunately.

So I suggest to add a fallback:

```scala
s.progress.map(p => BspCompileProgress(p, task))
  .getOrElse(BspCompileProgress(task))
```